### PR TITLE
Update platform channel page to new Android APIs and various cleanups

### PR DESCRIPTION
**Pull request by xster**
https://github.com/flutter/flutter/issues/46855
Fixes https://github.com/flutter/website/issues/3412

### CODE REVIEW

**Comment by xster** [comment]
Actually, it looks like Matt's out for the year

**Comment by xster** [comment]
Ummm looks like it built https://travis-ci.org/flutter/website/jobs/626844022?utm_medium=notification&utm_source=github_status but isn't showing for me on https://flutter.dev/docs/development/platform-integration/platform-channels. Was something else supposed to happen?

**Comment by sfshaza2** [comment]
It looks updated to me, @xster. Do you still see a problem?

**Comment by xster** [comment]
Ah it looks right to me too. I guess there's some invisible propagation time needed after Travis finished pushing to firebase.

File: src/docs/development/platform-integration/platform-channels.md [hunk 1]
```diff
@@ -185,32 +187,31 @@ interface that displays the battery state in a string,
...
+### Step 3: Add an Android platform-specific implementation
 
-### Step 3a: Add an Android platform-specific implementation using Java {#example-java}
-
-*Note*: The following steps use Java. If you prefer Kotlin, skip to step
-3b.
-
+{% samplecode android-channel %}
+{% sample Java %}
 Start by opening the Android host portion of your Flutter app in Android Studio:
```
Comment (matthew-carroll) [inline src/docs/development/platform-integration/platform-channels.md]: It looks like you're including plain text in a code block. Is that what you want?

File: src/docs/development/platform-integration/platform-channels.md [hunk 2]
```diff
@@ -119,10 +119,10 @@ Start by creating a new app:
 
 * In a terminal run: `flutter create batterylevel`
 
-By default our template supports writing Android code using Java, or iOS code
-using Objective-C. To use Kotlin or Swift, use the `-i` and/or `-a` flags:
+By default our template supports writing Android code using Kotlin, or iOS code
```
Comment (matthew-carroll) [inline src/docs/development/platform-integration/platform-channels.md]: It looks like this change removes Java examples. I know Kotlin is the default, but should we really remove the Java examples? We provided side-by-side Java/Kotlin examples elsewhere. Is there a reason why this guide should be different?

Comment (xster) [reply src/docs/development/platform-integration/platform-channels.md]: I didn't remove java. It's under a tab.

**Pull request merged**